app/bundles/CampaignBundle/Tests/Executioner/ScheduledExecutionerFunctionalTest.php

```diff
@@ -0,0 +1,448 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CampaignBundle\Tests\Executioner;
+
+use Mautic\CampaignBundle\Entity\Campaign;
+use Mautic\CampaignBundle\Entity\Event;
+use Mautic\CampaignBundle\Entity\Lead as CampaignLead;
+use Mautic\CampaignBundle\Entity\LeadEventLog;
+use Mautic\CampaignBundle\Executioner\ContactFinder\Limiter\ContactLimiter;
+use Mautic\CampaignBundle\Executioner\ScheduledExecutioner;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\Lead;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+final class ScheduledExecutionerFunctionalTest extends MauticMysqlTestCase
+{
+    public function testEventsAreExecuted(): void
+    {
+        $campaign = $this->createCampaign();
+        $event1   = $this->createEvent($campaign, 'Event 1');
+        $event2   = $this->createEvent($campaign, 'Event 2');
+        $contact  = $this->createContact();
+
+        $log1 = $this->createScheduledLog($event1, $contact, null, 1);
+        $log2 = $this->createScheduledLog($event1, $contact, null, 2);
+        $log3 = $this->createScheduledLog($event2, $contact, null, 3);
+        $log4 = $this->createScheduledLog($event2, $contact, null, 4);
+
+        $this->em->persist($campaign);
+        $this->em->persist($event1);
+        $this->em->persist($event2);
+        $this->em->persist($contact);
+        $this->em->persist($log1);
+        $this->em->persist($log2);
+        $this->em->persist($log3);
+        $this->em->persist($log4);
+        $this->em->flush();
+
+        /** @var ScheduledExecutioner $scheduledExecutioner */
+        $scheduledExecutioner = self::getContainer()->get('mautic.campaign.executioner.scheduled');
```

Comment:
By using `assert` the type gets actually verified in the test and dev environments compared to an annotation.
```suggestion
        $scheduledExecutioner = self::getContainer()->get('mautic.campaign.executioner.scheduled');
        \assert($scheduledExecutioner instanceof ScheduledExecutioner);
```

Perhaps you could get this service in the `setUp` method as it's needed in other tests in this file.

Reply:
Done.